### DISCUSSION TITLE
Specular usage

### DISCUSSION THREAD

**Discussion by ThomasGorisse** [google/filament#5416] [Q&A]
Hi,

I'm asking myself why are you using the specular filter for HDR cubemap in `IndirectLight` reflexion.
https://github.com/google/filament/blob/daff8fea9f828e1eca5a8f486dd6102552da5493/android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt#L212
As expected with this usage specular parts of objects material are not rendered.

Is it because I'm in a special case where no skybox is set ? Are the specular reflections coming from the `Scene` `Skybox` or `IndirectLight` ?

Thanks

**Comment by romainguy** [comment]
You're in a special case where no skybox is set. An IBL is just a skybox that's been filtered.

**Reply by ThomasGorisse** [reply]
Thanks.
So my best choice is to go for a non filtered IBL if you confirm that there is no way to use a "transparent" Skybox that will only generate the reflections?

**Reply by romainguy** [reply]
I don't understand what you are trying to do. You can run the prefilter and use that as an IBL without using a Skybox. In this case you'd have specular reflections and no background.

**Reply by ThomasGorisse** [reply]
Even if the case where a skybox is defined within an AR usage should really be more considered IMAO since it makes Filament become a mobile VR Engine with awesome results.

**Reply by romainguy** [reply]
I still don't understand what is the issue you are discussing. The skybox object is just used to render a background.

**Reply by ThomasGorisse** [reply]
The base question was only
"Where are the rendered model specular reflections coming from in this case since the specular filter is applied to the indirect light reflections ?"
https://github.com/google/filament/blob/main/android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt#L197-L227

**Reply by romainguy** [reply]
This code does what you normally do offline with cmgen, that’s all. Specular reflections come from the IndirectLight, and are computed by filtering an environment map, often the same texture as the one used for the skybox.

**Reply by ThomasGorisse** [reply]
And the "VR" usage I was speaking of looks like this:
https://user-images.githubusercontent.com/6597529/162608453-6dad85d0-677e-49aa-b47f-ec45b91bc0d5.mp4

**Reply by ThomasGorisse** [reply]
I think that the confusion comes from the fact that I'm wrong about the HDR cubemap given by ARCore.
I understood from the doc that it was not specular prefiltered but that might not be the case.
When setting the ARCore cubemap to the Skybox:
- Specular filter applied to the indirect light:
![Screenshot_20220410-110355.png](https://user-images.githubusercontent.com/6597529/162611186-3d1bec18-9a67-4755-9c20-7e5ee42c5c54.png)
- Direct ARCore HDR image to Filament indirect light (no specular filter)
![Screenshot_20220410-110408.png](https://user-images.githubusercontent.com/6597529/162611146-37be0499-9e93-41f9-b9ed-930743c26f73.png)
